test/ranges-test.cc

```diff
@@ -65,6 +65,29 @@ TEST(ranges_test, format_set) {
             "{\"one\", \"two\"}");
 }
 
+namespace adl {
+struct box {
+  int value;
+};
+
+auto begin(box& b) noexcept -> int* { return std::addressof(b.value); }
+
+auto end(box& b) noexcept -> int* { return std::addressof(b.value) + 1; }
+
+auto begin(const box& b) noexcept -> const int* {
+  return std::addressof(b.value);
+}
```

Comment:
noexcept is unnecessary, this is just a test. Also let's replace addressof with `&` for simplicity.